presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java

```diff
@@ -87,7 +91,8 @@ else if (keyType.getJavaType() == Slice.class) {
             methodHandle = METHOD_HANDLE_OBJECT;
         }
         methodHandle = MethodHandles.insertArguments(methodHandle, 0, legacyMissingKey);
-        methodHandle = methodHandle.bindTo(keyEqualsMethod).bindTo(keyType).bindTo(valueType);
+        FunctionInvoker functionInvoker = new FunctionInvoker(functionRegistry);
```

Comment:
Ideally, we should be able to declare an optional dependency on the CAST operator for the element type instead of using FunctionInvoker. But, this requires support from the framework, so this is ok for now.